src/main/java/luke/task/Task.java

```diff
@@ -0,0 +1,24 @@
+package luke.task;
+
+public abstract class Task {
+    protected String name;
+    protected Boolean isDone;
+
+    public Task(String taskName, boolean isDone) {
+        this.name = taskName;
+        this.isDone = isDone;
+    }
+
+    public void changeMark() {
+        this.isDone = !isDone;
+    }
+
+    public String showMark() {
+        return isDone
+                ? "[X]"
+                : "[ ]";
+    }
+
```

Comment:
Good idea to create abstract function to ensure the children class implement the functions!